src/EnergyPlus/PollutionModule.hh

```diff
@@ -77,336 +140,160 @@ namespace PollutionModule {
+        Constant::eFuel::FuelOilNo2, // FuelOil2
+        Constant::eFuel::Coal, // Coal
+        Constant::eFuel::Gasoline, // Gasoline
+        Constant::eFuel::Propane, // Propane
+        Constant::eFuel::Diesel, // Diesel
+        Constant::eFuel::OtherFuel1, // OtherFuel1
+        Constant::eFuel::OtherFuel2 // OtherFuel2
+    };
 
-        // Member Constructor
-        ComponentProps(int const FuelFactorType,
-                       Real64 const Source,
-                       Real64 const CO2Pollution,
-                       Real64 const COPollution,
-                       Real64 const CH4Pollution,
-                       Real64 const NOxPollution,
-                       Real64 const N2OPollution,
-                       Real64 const SO2Pollution,
-                       Real64 const PMPollution,
-                       Real64 const PM10Pollution,
-                       Real64 const PM25Pollution,
-                       Real64 const NH3Pollution,
-                       Real64 const NMVOCPollution,
-                       Real64 const HgPollution,
-                       Real64 const PbPollution,
-                       Real64 const WaterPollution,
-                       Real64 const NucHiPollution,
-                       Real64 const NucLoPollution)
-            : FuelFactorType(FuelFactorType), Source(Source), CO2Pollution(CO2Pollution), COPollution(COPollution), CH4Pollution(CH4Pollution),
-              NOxPollution(NOxPollution), N2OPollution(N2OPollution), SO2Pollution(SO2Pollution), PMPollution(PMPollution),
-              PM10Pollution(PM10Pollution), PM25Pollution(PM25Pollution), NH3Pollution(NH3Pollution), NMVOCPollution(NMVOCPollution),
-              HgPollution(HgPollution), PbPollution(PbPollution), WaterPollution(WaterPollution), NucHiPollution(NucHiPollution),
-              NucLoPollution(NucLoPollution)
-        {
-        }
+    constexpr std::array<std::string_view, (int)PollFuel::Num> pollFuelNamesUC = {
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::Electricity]], // Electricity
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::NaturalGas]], // NaturalGas
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::FuelOil1]], // FuelOil1
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::FuelOil2]], // FuelOil2
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::Coal]], // Coal
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::Gasoline]], // Gasoline
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::Propane]], // Propane
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::Diesel]], // Diesel
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::OtherFuel1]], // OtherFuel1
+        Constant::eFuelNamesUC[(int)pollFuel2fuel[(int)PollFuel::OtherFuel2]] // OtherFuel2
     };
 
-    struct CoefficientProps
+    enum class PollFuelComponent
```

Comment:
Yet another Fuel list, this one also doesn't have District but does have ElectricityPurchased and ElectricitySold. Because.